Feature/80 implement quiz events crud operations

RubyAtUsyd opened the pull request:
## Event APIs Implemented

1. **GET /api/v1/events?orgId={guid}&pageNumber={int}&pageSize={int}** - Retrieve paginated list of events for an organization.

2. **GET /api/v1/events/{eventId}** - Retrieve detailed information of a single event by ID.

3. **PUT /api/v1/events/{eventId}** - Update an existing event with partial update support, organization permission validation, and timestamp.

4. **DELETE /api/v1/events/{eventId}** - Delete an event with organization permission check and returns 204 No Content on success.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* New Features
  * Full event CRUD: paginated listing, get-by-id, update and delete with organization-scoped permission checks, validation, partial updates, and creator info.
* Refactor
  * Event status renamed "Live" → "Published" and now serialized as strings (may affect integrations).
* Documentation
  * Removed the Event Creation API requirements document.
* Chores
  * Removed local API debugging script.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

RubyAtUsyd committed:
feat(dto): add GetEventListResponse models for event list API

RubyAtUsyd committed:
feat(repository): add pagination method to IEventRepository interface

RubyAtUsyd committed:
feat(repository): implement pagination in EventRepository

RubyAtUsyd committed:
feat(service): add GetEventListAsync method to IEventService interface

RubyAtUsyd committed:
feat(service): implement GetEventListAsync method in EventService

RubyAtUsyd committed:
feat(api): add GET endpoint for event list with pagination

RubyAtUsyd committed:
feat(dto): add GetEventByIdResponse model

RubyAtUsyd committed:
feat(service): add GetEventByIdAsync method to IEventService interface

RubyAtUsyd committed:
feat(service): implement GetEventByIdAsync method in EventService

RubyAtUsyd committed:
feat(api): add GET endpoint for event by ID

RubyAtUsyd committed:
feat: Update EventStatus enum to support multiple formats

- Add JsonStringEnumConverter to support string and numeric values
- Rename 'Live' to 'Published' for better clarity
- Assign explicit numeric values: Draft=0, Published=1, Ended=2
- Support case-insensitive string parsing (e.g., 'draft', 'Draft', 'DRAFT')
- Support numeric values (0, 1, 2) for API flexibility

RubyAtUsyd committed:
feat: Add Update Event DTOs and Repository layer

- Create UpdateEventRequest DTO with optional fields for partial updates
- Create UpdateEventResponse DTO to return updated event details
- Add UpdateAsync method to IEventRepository interface
- Implement UpdateAsync method in EventRepository

RubyAtUsyd committed:
feat: Implement Update Event Service layer

- Add UpdateEventAsync method to IEventService interface
- Implement UpdateEventAsync in EventService with:
  - Event existence validation
  - User permission check (same organization)
  - Partial update support (only update provided fields)
  - Business rule validation for time range
  - Automatic UpdatedAt timestamp update
- Add GetUserOrganizationIdAsync helper method
- Inject ApplicationDbContext for user lookup

RubyAtUsyd committed:
feat: Add Update Event API endpoint

- Add PUT /api/v1/events/{eventId} endpoint
- Implement authorization check (extract userId from JWT)
- Add proper HTTP status codes (200, 400, 401, 403, 404)
- Add Swagger documentation with response types

RubyAtUsyd committed:
fix: Use DateTimeOffset for Update Event DTOs

- Change DateTime to DateTimeOffset in UpdateEventRequest
- Change DateTime to DateTimeOffset in UpdateEventResponse
- Update EventService to use DateTimeOffset.UtcNow
- Match Event entity type definitions

RubyAtUsyd committed:
feat: Add Delete Event Repository layer

- Add DeleteAsync method to IEventRepository interface
- Implement DeleteAsync in EventRepository
- Use EF Core Remove method for soft/hard delete

RubyAtUsyd committed:
feat: Implement Delete Event Service layer

- Add DeleteEventAsync method to IEventService interface
- Implement DeleteEventAsync in EventService with:
  - Event existence validation
  - User permission check (same organization)
  - Structured logging for audit trail

RubyAtUsyd committed:
feat: Add Delete Event API endpoint

- Add DELETE /api/v1/events/{eventId} endpoint
- Implement authorization check (extract userId from JWT)
- Return 204 No Content on successful deletion
- Add proper HTTP status codes (204, 401, 403, 404)
- Add Swagger documentation

RubyAtUsyd committed:
fix:remove unused files

joeqiwx commented on src/MeetlyOmni.Api/Service/EventService/EventService.cs:
```diff
@@ -89,6 +200,141 @@ public async Task<CreateEventResponse> CreateEventAsync(
+        if (request.EndTime.HasValue)
+        {
+            existingEvent.EndTime = request.EndTime.Value;
+        }
```
Could we use mapper or other tools to avoid these if checks

RubyAtUsyd replied:
Yes, I use automapper to handle this checks now
`   _mapper.Map(request, existingEvent);
    existingEvent.UpdatedAt = DateTimeOffset.UtcNow;
    var updatedEvent = await _eventRepository.UpdateAsync(existingEvent, cancellationToken);
    `

joeqiwx commented on src/MeetlyOmni.Api/Service/EventService/EventService.cs:
```diff
@@ -18,14 +21,122 @@ namespace MeetlyOmni.Api.Service.EventService;
+        {
+            throw new DomainValidationException(
+                new Dictionary<string, string[]> { { "PageSize", new[] { "Page size must be between 1 and 100." } } });
+        }
```
Could we put the pageNumber and pageSize in a dto and then use the DataAnnotations e.g. range to do the check

RubyAtUsyd replied:
Yes, now I moved pageNumber and pageSize to dto class named GetEventListRequest， and initiate the validation rules in the dto.



> Could we put the pageNumber and pageSize in a dto and then use the DataAnnotations e.g. range to do the check

joeqiwx commented on src/MeetlyOmni.Api/Service/EventService/EventService.cs:
```diff
@@ -89,6 +200,141 @@ public async Task<CreateEventResponse> CreateEventAsync(
+        }
+
+        // Validate user has permission to update this event (belongs to same organization)
+        var userOrgId = await GetUserOrganizationIdAsync(userId, cancellationToken);
```
Could we use the IAuthorizationService to handle this check before we call the service

RubyAtUsyd replied:
Yes, I add a authorization middleware called by the controller to handle the check and this is indepent and reusable.

RubyAtUsyd committed:
refactor: Use AutoMapper for Update Event partial updates

RubyAtUsyd committed:
refactor: Use DTO with DataAnnotations for pagination validation

- Create GetEventListRequest DTO with Range validation
- Remove manual if-checks for pageNumber and pageSize
- Let ASP.NET Core automatically validate query parameters
- Update IEventService and EventController signatures

RubyAtUsyd committed:
refactor: Use IAuthorizationService for organization permission checks

- Create SameOrganizationRequirement and Handler
- Add IOrganizationResource interface for resources with OrgId
- Event entity implements IOrganizationResource
- Move authorization checks from Service to Controller layer
- Remove userId parameter from Update/Delete methods
- Remove GetUserOrganizationIdAsync helper method
- Follow ASP.NET Core authorization best practices

RubyAtUsyd committed:
fix: Add GetEventEntityByIdAsync for authorization checks

- Add GetEventEntityByIdAsync method to return Event entity
- Update Controller to use entity (not DTO) for authorization
- Fix org_id claim name in authorization handler
- All authorization tests passing

RubyAtUsyd committed:
fix: Restore authorization handlers and improve Antiforgery config formatting

- Restore SameOrganizationAuthorizationHandler registration for Event authorization
- Format Antiforgery cookie configuration for better readability
- Ensure proper environment-based SecurePolicy and SameSite settings
- Fix trailing whitespace in authorization handler registration
